CMakeLists.txt

```diff
@@ -69,12 +68,15 @@ endif()
 if( BUILD_FAT_LIBMLIRMIOPEN )
   set(BUILD_SHARED_LIBS OFF CACHE BOOL "")
   set(LLVM_BUILD_LLVM_DYLIB OFF CACHE BOOL "")
+  set(LLVM_INSTALL_TOOLCHAIN_ONLY ON CACHE BOOL "")
+  set(USE_ROCM_4_4_OR_OLDER 0 CACHE BOOL "Make libMLIRMIOpen runtime independent")
```

Comment:
This doesn't feel right. That is, why is ROCm version dependent on whether we are building a fat libMLIRMIOpen?

(The toolchain install is probably OK)

Reply:
`USE_ROCM_4_4_OR_OLDER ` is a poor name. It is somewhat rocm version dependent but in reality it controls whether or not we link with hip.

Reply:
In which case, could you give it a rename here?

Reply:
@krzysz00 Well I'm not sure just renaming it is the best approach as well because this remind us to get rid of the cmake variable when we are above ROCm 4.4. I think that was the intention, right? @yiqian1 

Since this cmake variable is doomed to be a short-term solution, I feel like we should just keep it as-is.

Reply:
Good point. In which case, could you toss in a comment explaining why this is being set to 0 on BUILD_FAT_LIBMLIRMIOPEN, since that'll be confusing when someone looks at it?

(also, is not linking with HIP the right call here?)

Reply:
That's fair, addressed in 39c8a2f.

And yes, not linking with hip make the fat lib runtime dependent. Our client can rely on opencl instead of hip.